go/vt/sqlparser/truncate_query.go

```diff
@@ -17,17 +17,28 @@ limitations under the License.
 package sqlparser
 
 import (
-	"flag"
+	"github.com/spf13/pflag"
+
+	"vitess.io/vitess/go/vt/servenv"
 )
 
 var (
 	// TruncateUILen truncate queries in debug UIs to the given length. 0 means unlimited.
-	TruncateUILen = flag.Int("sql-max-length-ui", 512, "truncate queries in debug UIs to the given length (default 512)")
+	TruncateUILen int
 
 	// TruncateErrLen truncate queries in error logs to the given length. 0 means unlimited.
-	TruncateErrLen = flag.Int("sql-max-length-errors", 0, "truncate queries in error logs to the given length (default unlimited)")
+	TruncateErrLen int
 )
 
+func registerQueryTruncationFlags(fs *pflag.FlagSet) {
+	fs.IntVar(&TruncateUILen, "sql-max-length-ui", 512, "truncate queries in debug UIs to the given length (default 512)")
+	fs.IntVar(&TruncateErrLen, "sql-max-length-errors", 0, "truncate queries in error logs to the given length (default unlimited)")
+}
+
+func init() {
+	servenv.OnParse(registerQueryTruncationFlags)
```

Comment:
did you mean to install this on _every_ binary?

Reply:
Yes, the function that uses these flags is actually in truncating the logs. So any binary which receives an SQLError and tries to log it will be calling these functions and will need these flags. I ran the command that the merger-guide lists, and these flags are currently present on all the binaries - 
```
 ~/vitess   pflags-sqlparser-fix  go list -f '{{ .ImportPath }}| {{ join .Deps " " }} ' ./go/cmd/... | awk -F"|" '$2 ~ " vitess.io/vitess/go/vt/sqlparser " {print $1}'
vitess.io/vitess/go/cmd/mysqlctl
vitess.io/vitess/go/cmd/mysqlctld
vitess.io/vitess/go/cmd/query_analyzer
vitess.io/vitess/go/cmd/rulesctl
vitess.io/vitess/go/cmd/rulesctl/cmd
vitess.io/vitess/go/cmd/rulesctl/common
vitess.io/vitess/go/cmd/topo2topo
vitess.io/vitess/go/cmd/vtaclcheck
vitess.io/vitess/go/cmd/vtadmin
vitess.io/vitess/go/cmd/vtbackup
vitess.io/vitess/go/cmd/vtbench
vitess.io/vitess/go/cmd/vtclient
vitess.io/vitess/go/cmd/vtcombo
vitess.io/vitess/go/cmd/vtctl
vitess.io/vitess/go/cmd/vtctld
vitess.io/vitess/go/cmd/vtctldclient
vitess.io/vitess/go/cmd/vtctldclient/cli
vitess.io/vitess/go/cmd/vtctldclient/command
vitess.io/vitess/go/cmd/vtctldclient/docgen
vitess.io/vitess/go/cmd/vtexplain
vitess.io/vitess/go/cmd/vtgate
vitess.io/vitess/go/cmd/vtgateclienttest
vitess.io/vitess/go/cmd/vtgr
vitess.io/vitess/go/cmd/vtorc
vitess.io/vitess/go/cmd/vttablet
vitess.io/vitess/go/cmd/vttestserver
vitess.io/vitess/go/cmd/zk
```

Reply:
sure, those _import_ the package, but for example (you can see several of my PRs that describe this), `cmd/zk` imports `go/vt/vtctl` but only to call `DecodeContent`, which doesn't use this flag's value, so it should not be registered or in the help/usage for that command.

unfortunately you need to actually trace the import dependencies to see if "imports the flag" means "needs the flag" or not; this command is just the launch-point for that analysis.

Reply:
(i think that anything with a `/debug/*` http endpoint is probably the one that needs this, with the exception of vtadmin which is doing it's own thing - but you would need to trace the code, in particular how it uses `go/vt/vtexplain`)

Reply:
That is what I thought in the beginning too, but it's not the UI that is troublesome, its the logging part. `TruncateForLog` gets called in `Error` function of `SQLError`. So any binary which can receive a SQLError will need the flags, because it is used in printing the Error. So, `vtgate`, `vtcombo`, `vttablet`, `vtctld`, `vtctl` definitely need it. I am not sure how to check if the others need it or not, because there is not good way to track if they'll receive SQL error in any of there functions. How do you suggest we proceed?
The first way was to add the flag to everything importing it, but that seems too much. The other way is to only add it to the five binaries listed above. Incase any other binary is using it, it they'll just use the default values, but then we'll have to rely on users to tell us if some binary actually needs it and we didn't add it

Reply:
Okay, @ajm188 following your advice, I looked at all the commands that were receiving a `SQLError` or trying to convert to it after receiving a string error message and I think I have a list of binaries that use the flags - 
```
"vtgate"
"vttablet"
"vtcombo"
"vtctld"
"vtctl"
"vtexplain"
"vtbackup"
"vttestserver"
"vtbench"
```

Pushed changes accordingly